Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -133,17 +133,19 @@ class TopSitesPanel: UIViewController {
         collection?.indexPathsForVisibleItems().forEach(updateRemoveButtonStateForIndexPath)
     }
 
-    private func deleteTileForSuggestedSite(site: SuggestedSite) -> Success {
+    private func deleteTileForSuggestedSite(site: Site) -> Success {
         var deletedSuggestedSites = profile.prefs.arrayForKey("topSites.deletedSuggestedSites") as! [String]
-        deletedSuggestedSites.append(site.url)
+        if !deletedSuggestedSites.contains(dataSource.extractDomainURL(site.url)) {
```

Comment:
Is this check necessary? If we've already deleted a site, we shouldn't show a tile for that domain anymore, so how would we get into a situation where we'd be deleting it again?